sklearn/cluster/tests/test_birch.py

```diff
@@ -49,26 +51,31 @@ def test_partial_fit():
     assert_array_equal(brc_partial.subcluster_labels_, brc.subcluster_labels_)
 
 
-def test_birch_predict():
+def test_birch_predict(global_dtype):
     # Test the predict method predicts the nearest centroid.
     rng = np.random.RandomState(0)
     X = generate_clustered_data(n_clusters=3, n_features=3, n_samples_per_cluster=10)
+    X = X.astype(global_dtype)
 
     # n_samples * n_samples_per_cluster
     shuffle_indices = np.arange(30)
     rng.shuffle(shuffle_indices)
     X_shuffle = X[shuffle_indices, :]
     brc = Birch(n_clusters=4, threshold=1.0)
     brc.fit(X_shuffle)
+    # TODO: the subcluster_centers_ fitted attribute must have the same dtype
+    # assert brc.subcluster_centers_.dtype == global_dtype
+
```

Comment:
The TODO is already known from the meta issue of transformers. I'd rather not add the dtype check in this test. I think each PR enabling dtype preservation for a transformer should come with a dedicated test for the checking of the dtype of the appropriate attributes to check.
```suggestion
```